packages/react/src/NavList/NavList.tsx

```diff
@@ -290,6 +291,31 @@ const Group: React.FC<NavListGroupProps> = ({title, children, sx: sxProp = defau
 
 Group.displayName = 'NavList.Group'
 
+export type NavListGroupHeadingProps = ActionListGroupHeadingProps
+
+/**
+ * This is an alternative to the `title` prop on `NavList.Group`.
+ * It was primarily added to allow links in group headings.
+ */
+const GroupHeading: React.FC<NavListGroupHeadingProps> = ({sx: sxProp = defaultSxProp, ...rest}) => {
+  return (
+    <ActionList.GroupHeading
+      as="h3"
```

Comment:
(nit) optional suggestion: 
from line 285: 
> TODO: API update to give flexibility to NavList.Group title's heading level 

Maybe this is a good opportunity to do that here? (take `as` as a prop that defaults to `h3` and can be overriden if necessary). We could then use this same component for the `title` prop instead of `ActionList.GroupHeading` as well 🤔

Wouldn't make the `title`'s prop heading level itself configurable, but might offer an alternative manually through the new `GroupHeading`